app/src/types.ts

```diff
@@ -100,11 +102,28 @@ interface Description {
   custom: Record<string, any> | null;
 }
 
-interface Artefact {
+export interface Artefact {
   description: string;
   paths: string[];
 }
 
+export interface Role {
+  path: string;
+  role: Roles;
+}
+
+export enum Roles {
+  Orderly = "orderly",
+  Resource = "resource",
+  Dependency = "dependency",
+  Shared = "shared",
+}
+
+export interface SharedResource {
+  here: string;
+  there: string;
+}
+
```

Comment:
"Role" is a really confusing type here, given that we have permission-related roles as well. It's more verbose, but could we call these something more precise like "OrderlyInputRole"..?

Reply:
Good point